### PR TITLE
chore: add `--diff` flag in svc/job/env package 

### PR DESCRIPTION
**Pull request by Lou1415926**
Large PR but the real logic is repetitive across the three commands. So consider #additions as #additions/3 additions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/env_package.go [hunk 1]
```diff
@@ -131,6 +134,14 @@ func newPackageEnvOpts(vars packageEnvVars) (*packageEnvOpts, error) {
 
 // Validate returns an error for any invalid optional flags.
 func (o *packageEnvOpts) Validate() error {
+	if o.showDiff {
+		if o.outputDir != "" {
+			return fmt.Errorf("`--%s` cannot be specified together with `--%s`", diffFlag, stackOutputDirFlag)
+		}
```
Comment (efekarakus) [inline internal/pkg/cli/env_package.go]: can we use `cmd.MarkFlagsMutuallyExclusive` instead to get rid of the functionality? (samesies in other commands)

File: internal/pkg/cli/flag.go [hunk 2]
```diff
@@ -249,6 +250,7 @@ const (
 	yesFlagDescription          = "Skips confirmation prompt."
 	resourceTagsFlagDescription = `Optional. Labels with a key and value separated by commas.
 Allows you to categorize resources.`
+	diffFlagDescription = "Show the diff of the to-be-deployed CFN template against the deployed one."
```
Comment (efekarakus) [inline internal/pkg/cli/flag.go]: To go with a similar wording to `cdk diff`, perhaps:
> Compares the specified stack with the deployed stack or a local template file
```suggestion
	diffFlagDescription = "Compares the generated CloudFormation template to the deployed stack."
```

**Review by efekarakus — COMMENTED** [review]
A nice thing that I've read in the CDK diff command is this:
> and returns with status 1 if any difference is found

So I assume it exits with 0 if there are no changes. We should consider also supporting hte same behavior. But the changes look great, just two small suggestions.

File: internal/pkg/cli/svc_package.go [hunk 3]
```diff
@@ -470,5 +488,9 @@ func buildSvcPackageCmd() *cobra.Command {
 	cmd.Flags().StringVar(&vars.tag, imageTagFlag, "", imageTagFlagDescription)
 	cmd.Flags().StringVar(&vars.outputDir, stackOutputDirFlag, "", stackOutputDirFlagDescription)
 	cmd.Flags().BoolVar(&vars.uploadAssets, uploadAssetsFlag, false, uploadAssetsFlagDescription)
+	cmd.Flags().BoolVar(&vars.showDiff, diffFlag, false, diffFlagDescription)
+
+	cmd.MarkFlagsMutuallyExclusive(diffFlag, stackOutputDirFlag)
+	cmd.MarkFlagsMutuallyExclusive(diffFlag, uploadAssetsFlag)
```
Comment (iamhopaul123) [inline internal/pkg/cli/svc_package.go:482]: Why do we not allow them to specify upload assets flag?

**Review by efekarakus — APPROVED** [review]

**Review by dannyrandall — APPROVED** [review]

File: internal/pkg/cli/svc_package.go (hunk 3 above)
Comment (Lou1415926) [reply internal/pkg/cli/svc_package.go:482]: `copilot svc package --diff --upload-assets`  - not sure if it makes sense to upload assets while they just want to show diff 🤔  

We upload env file, custom resource and addons with `--upload-assets`. Env files URLs are passed via. `Parameters` iirc, custom resources/addons URLs are deterministic and are rendered into the template regardless whether they are uploaded. Therefore, with/without `--upload-assets` should render the same template. We can definitely relax this constraint now or later, I am not feeling strong on this.

**Pull request merged**